### ISSUE TITLE
[Bug]: TextInput with no helper text uses invalid aria-describedby 

### ISSUE THREAD

**Issue by dcwarwick** [carbon-design-system/carbon#10107]
### Package

carbon-components-react

### Package version

7.47.2

### Description

With #9873, created by @dakahn to address #9630, an `aria-describedby` has been added to the TextInput generated DOM, set to the id of a helper text DIV. However, the helper text DIV is only generated *if* the `helperText` prop has been set. This means that when the `helperText` prop is NOT set the generated DOM includes an `aria-describedby` which contains an id that doesn't exist.

When `aria-describedby` contains an id that doesn't exist, this is an error, and causes accessibility testing tools to report violations whenever TextInput components without `helperText` set are used. For example, `accessibility-checker` v3.1.11 reports violations as follows:
![image](https://user-images.githubusercontent.com/6385315/141538244-42a94371-b5cc-4ddd-b3a2-ce8f38d9c199.png)
The above example is from the ExportModal test suite of @carbon/ibm-cloud-cognitive. This test has been temporarily skipped in order to allow us to continue work until this issue is fixed.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
